modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -3447,6 +3462,12 @@ private CompletableFuture<Boolean> ensureReplicaIsPrimary(ReplicaRequest request
                             );
                         }
 
+                        synchronized (this) {
+                            if (leaseStartTime < primaryReplicaMeta.getStartTime().longValue()) {
```

Comment:
How do you guarantee that command will be filled with matched leaseStartTime that was assosiated with the command on enusreReplicaIsPrimary? Meanging. 
1. request1 -> ensureReplicaIsPrimary -> leaseStartTime = t0 -> lock -> hang 
2. request2 -> ensureReplicaIsPrimary -> leaseStartTime = t100 -> ...
3. reqest1 -> wakes up -> command(t100)

Reply:
fixed